plugin/src/main/java/org/opensearch/ml/action/deploy/TransportDeployModelOnNodeAction.java

```diff
@@ -146,42 +144,51 @@ private MLDeployModelNodeResponse createDeployModelNodeResponse(MLDeployModelNod
-                .requestType(MLForwardRequestType.DEPLOY_MODEL_DONE)
-                .taskId(taskId)
-                .modelId(modelId)
-                .workerNodeId(clusterService.localNode().getId())
-                .build();
-            MLForwardRequest deployModelDoneMessage = new MLForwardRequest(mlForwardInput);
-
-            transportService
-                .sendRequest(
-                    getNodeById(coordinatingNodeId),
-                    MLForwardAction.NAME,
-                    deployModelDoneMessage,
-                    new ActionListenerResponseHandler<MLForwardResponse>(taskDoneListener, MLForwardResponse::new)
-                );
-        }, e -> {
-            MLForwardInput mlForwardInput = MLForwardInput
-                .builder()
-                .requestType(MLForwardRequestType.DEPLOY_MODEL_DONE)
-                .taskId(taskId)
-                .modelId(modelId)
-                .workerNodeId(clusterService.localNode().getId())
-                .error(MLExceptionUtils.getRootCauseMessage(e))
-                .build();
-            MLForwardRequest deployModelDoneMessage = new MLForwardRequest(mlForwardInput);
-
-            transportService
-                .sendRequest(
-                    getNodeById(coordinatingNodeId),
-                    MLForwardAction.NAME,
-                    deployModelDoneMessage,
-                    new ActionListenerResponseHandler<MLForwardResponse>(taskDoneListener, MLForwardResponse::new)
-                );
-        }));
+        deployModel(
+            modelId,
+            modelContentHash,
+            mlTask.getFunctionName(),
+            localNodeId,
+            coordinatingNodeId,
+            deployToAllNodes,
+            mlTask,
+            ActionListener.wrap(r -> {
+                MLForwardInput mlForwardInput = MLForwardInput
+                    .builder()
+                    .requestType(MLForwardRequestType.DEPLOY_MODEL_DONE)
+                    .taskId(taskId)
+                    .modelId(modelId)
+                    .workerNodeId(clusterService.localNode().getId())
+                    .build();
+                MLForwardRequest deployModelDoneMessage = new MLForwardRequest(mlForwardInput);
+
+                transportService
+                    .sendRequest(
+                        getNodeById(coordinatingNodeId),
+                        MLForwardAction.NAME,
+                        deployModelDoneMessage,
+                        new ActionListenerResponseHandler<>(taskDoneListener, MLForwardResponse::new)
+                    );
+            }, e -> {
+                MLForwardInput mlForwardInput = MLForwardInput
+                    .builder()
+                    .requestType(MLForwardRequestType.DEPLOY_MODEL_DONE)
+                    .taskId(taskId)
+                    .modelId(modelId)
+                    .workerNodeId(clusterService.localNode().getId())
+                    .error(MLExceptionUtils.getRootCauseMessage(e))
```

Comment:
Looks like for `onResponse` and `onFailure` we are doing mostly same except the error message: `.error(MLExceptionUtils.getRootCauseMessage(e))`

Can we add any comment to explain this?

Reply:
This is not introduced in this PR, we can skip this